src/rv/comm/rcssserver/GameState.java

```diff
@@ -48,6 +48,34 @@
         void gsServerMessageProcessed(GameState gs);
     }
 
+    public enum FoulType {
+        CROWDING(0, "crowding"), TOUCHING(1, "touching"), ILLEGALDEFENCE(2, "illegal defence"),
+        ILLEGALATTACK(3, "illegal attack"), INCAPABLE(4, "incapable"),
+        KICKOFF(5, "illegal kickoff"), CHARGING(6, "charging");
+
+        private int    idx;
+        private String str;
+
+        FoulType(int i, String s) {
+            this.idx = i;
+            this.str = s;
+        }
+
+        public String toString() {
+            return str;
+        }
+    }
+
+    public class Foul {
+
+        public float    time;
+        public int      index;
+        public FoulType type;
+        public int      team;
+        public int      unum;
```

Comment:
RoboViz seems to use `agentID` for the most part rather than `unum`, probably best to stay consistent with that.